jax/core.py

```diff
@@ -1623,3 +1630,39 @@ def initial_style_staging():
       yield
     finally:
       trace_state.initial_style = prev
+
+class Float0Array:
+  """Array used to represent the tangent values of int/bool primals.
+
+  No operations are defined for this array-type by design: these values can't be
+  used in combination with non-trivial vector spaces. No implicit dtype
+  conversion is done.  To use these values, the array can be cast to float32
+  zeros using the `to_zeros` method.
+  """
+  __slots__ = ['shape']
+  dtype = dtypes.float0
+
+  __array_priority__ = 100
+
+  # Float0Array has methods that are dynamically populated in lax_numpy.py,
+  # and this annotation is needed to make pytype happy.
+  _HAS_DYNAMIC_ATTRIBUTES = True
```

Comment:
The downside of this approach is that now pytype can't catch these errors at compile time (but I guess it would have hard time identifying when `Float0Array` is created, anyways).